### PR TITLE
WIP: Add `oc adm etcd` to support etcd cluster management

### PR DESCRIPTION
**Pull request by marun**

### CODE REVIEW

File: pkg/cli/admin/etcd/dnshelpers.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
+package etcd
```
Comment (hexfusion) [inline pkg/cli/admin/etcd/dnshelpers.go:1]: maybe it's time to consider these helpers in library-go?

File: pkg/cli/admin/etcd/listmembers.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	genericclioptions.IOStreams
+}
+
+func NewCommandListMembers(f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := ListMembersOptions{
+		IOStreams: streams,
+	}
+	cmd := &cobra.Command{
+		Use:   "list-members",
```
Comment (hexfusion) [inline pkg/cli/admin/etcd/listmembers.go:26]: do we want to align the naming with etcd client command? IE MemberList? I think it's fine to align with k8s naming standards if you feel that is more appropriate. But adm should not have to rethink common etcd commands.

**Review by hexfusion — COMMENTED** [review]
nice, having this will be a great step for etcd admin UX. Added a few notes on first pass.

File: pkg/cli/admin/etcd/client.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+	defer cancel()
+
+	endpoints, err := etcdEndpoints(ctx, kubeClient, configClient)
+	if err != nil {
+		return nil, err
+	}
+
+	// Read etcd client certs from the api
+	certSecretName := "etcd-client"
+	secret, err := kubeClient.CoreV1().Secrets(operatorNamespace).Get(ctx, certSecretName, metav1.GetOptions{})
```
[the remark this answers is not in the document]
Comment (marun) [reply pkg/cli/admin/etcd/client.go:47]: Done

File: pkg/cli/admin/etcd/client.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	}
+	for _, node := range nodes.Items {
+		internalIP, err := GetEscapedPreferredInternalIPAddressForNodeName(network, &node)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get internal IP for node: %w", err)
+		}
+		endpoints = append(endpoints, fmt.Sprintf("https://%s:2379", internalIP))
+	}
+
+	// Add ip address of bootstrap node it is still present
```
[the remark this answers is not in the document]
Comment (marun) [reply pkg/cli/admin/etcd/client.go:137]: Done

File: pkg/cli/admin/etcd/dnshelpers.go (hunk 1 above)
Comment (marun) [reply pkg/cli/admin/etcd/dnshelpers.go:1]: Maybe, or as a pre-req for this merging?

File: pkg/cli/admin/etcd/listmembers.go (hunk 2 above)
Comment (marun) [reply pkg/cli/admin/etcd/listmembers.go:26]: I would defer to @soltysh on this question.

**Pull request closed without merging**